programs/server/config.xml

```diff
@@ -66,69 +66,289 @@
-         See the openSSL section below.
-         See also 'interserver_http_credentials'.
+        <my_http_proxy>
+            <!-- Proxy information from X-Forwarded-For HTTP header.-->
+            <protocol>HTTP</protocol>
+
+            <!-- List of IPv4 and/or IPv6 networks, connections from which will be assumed to be from trusted proxies.
+                 Note, that all intermediate peers (but not the initiator client peer) from X-Forwarded-For must be from
+                  some of the networks from this list, for the information about the forwarding chain to be considered authentic.
+              -->
+            <trust>
+                <net>10.10.20.1/24</net>
+            <trust>
+        </my_http_proxy>
+
+    </proxies>
+
+
+    <!-- Interface definitions for ClickHouse server to listen for.
+         Each section can have arbitrary name but names must be unique.
+         The names will be used in logging, as keys for stats, etc.
+         Each section have one mandatory parameter - 'protocol'.
+         Supported protocols are:
+            - native_tcp
+            - native_http
+            - native_grpc
+            - mysql
+            - postgresql
+            - prometheus
+            - interserver_http
+            - keeper_tcp
+         You can define any number of interfaces for each of the protocols.
       -->
-    <!-- <interserver_https_port>9010</interserver_https_port> -->
+    <interfaces>
+
+        <!-- Interface for interactions using plain native TCP protocol with:
+               - clickhouse-client and other native ClickHouse tools (clickhouse-benchmark, clickhouse-copier);
+               - clickhouse-server with other clickhouse-servers for distributed query processing;
+               - ClickHouse drivers and applications supporting native protocol
+             See also 'secure_native_tcp' for secure connections.
+          -->
+        <plain_native_tcp>
+            <protocol>native_tcp</protocol>
+
+            <!-- Use :: (wildcard IPv6 address), if you want to accept connections both with IPv4 and IPv6 from everywhere.
+                 If you open connections from wildcard address, make sure that at least one of the following measures applied:
+                   - server is protected by firewall and not accessible from untrusted networks;
+                   - all users are restricted to subset of network addresses (see users.xml);
+                   - all users have strong passwords, only secure (TLS) interfaces are accessible, or connections are only made via TLS interfaces.
+                   - users without password have readonly access.
+                 See also: https://www.shodan.io/search?query=clickhouse
+              -->
+            <!--
+            <listen>
+                <host>127.0.0.1</host>
+                <host>::1</host>
+            </listen>
+            -->
+
+            <port>9000</port>
+            <enable_tls>no</enable_tls>
```

Comment:
is it mandatory?

Reply:
I don't like the name too much, some alternatives:
1) `<encrypted>1</encrypted>`
2) `<secure>1</secure>`
3) `<use_tls />` or just `<tls />`

AFAIR you can configure openssl to use SSL only (w/o TLS).

Reply:
> is it mandatory?

No. Just for clarity.